source/developers-guide/shopware-5-plugin-update-guide/index.md

```diff
@@ -110,163 +156,166 @@ The new files contain the following source code:
+    public function __construct(DBALConfigReader $configReader, $pluginName)
+    {
+        $this->configReader = $configReader;
+        $this->pluginName = $pluginName;
     }
 
-    public function onSecureDetailPostDispatch(Enlight_Event_EventArgs $arguments)
+    public static function getSubscribedEvents()
     {
-        /**@var $controller Shopware_Controllers_Frontend_Listing*/
-        $controller = $arguments->getSubject();
+        return [
+            'Enlight_Controller_Action_PostDispatchSecure_Frontend_Detail' => 'onPostDispatchFrontendDetail'
+        ];
+    }
 
-        $controller->View()->addTemplateDir($this->Path() . 'Views/common/');
+    public function onPostDispatchFrontendDetail(\Enlight_Event_EventArgs $args)
+    {
+        /** @var \Shopware_Controllers_Frontend_Detail $subject */
+        $subject = $args->get('subject');
 
-        if (Shopware()->Shop()->getTemplate()->getVersion() >= 3) {
-            $controller->View()->addTemplateDir($this->Path() . 'Views/responsive/');
-        } else {
-            $controller->View()->addTemplateDir($this->Path() . 'Views/emotion/');
-            $controller->View()->extendsTemplate('frontend/detail/example1.tpl');
-        }
+        $config = $this->configReader->getByPluginName($this->pluginName);
 
-        $controller->View()->mediaSelection = $this->Config()->mediaselection;
+        $subject->View()->assign('mediaSelection', $config['mediaSelection']);
     }
 }
 ```
 
-- The "Views/common/" directory is always registered as a template directory. Templates inside this directory are now loaded for both template versions.
-- A template version check was added, so that only the corresponding view directory is loaded.
-- Responsive - Template files for the responsive template loaded automatically over the inheritance by using the same names. Therefore, only the template directory must registered here
-- Emotion - In the emotion template we can load the template extension like in Shopware 4. After the template directory is registered, the template file is loaded via the __extendsTemplate__ call.
-
-__Notice: Do not use "_default/" or "_emotion/" at the beginning of the extend call. Use "parent:" instead.__
-
-__Notice: To allow other templates to easily extend your templates, you should provide your code in a separate file. Include that file in other templates, inside Smarty blocks, thus allowing different entry point.__
-
 ## Uninstall
 
 During the uninstall process, the user can now be prompted which data he wishes to remove.
-Existing __uninstall()__ method should remove all data, and the new __secureUninstall()__ method should only remove non-user related data.
+Existing __uninstall()__ method should remove all data. 
+Use the `UninstallContext` which contains the __keepUserData()__ method. If the return value is `true` you should only remove non-user related data.
 
-##### Bootstrap.php - How to use 'secureUninstall()'
+### How to use 'secureUninstall()'
```

Comment:
the secureUninstall method does not exist in the new plugin system

Reply:
D